consensus/quorum/one-node-staked-vote.go

```diff
@@ -39,31 +49,44 @@ func (v *stakedVoteWeight) Policy() Policy {
 
 // IsQuorumAchieved ..
 func (v *stakedVoteWeight) IsQuorumAchieved(p Phase) bool {
-	// TODO Implement this logic w/Chao
-	// soFar := numeric.ZeroDec()
+	t := numeric.NewDecFromBigInt(v.QuorumThreshold())
+	currentTotalPower := v.computeCurrentTotalPower(p)
+
+	utils.Logger().Info().
+		Str("policy", v.Policy().String()).
+		Str("phase", p.String()).
+		Str("threshold", t.String()).
+		Str("total-power-of-signers", currentTotalPower.String()).
+		Msg("Attempt to reach quorum")
+
+	return currentTotalPower.GT(t)
+}
+
+func (v *stakedVoteWeight) computeCurrentTotalPower(p Phase) numeric.Dec {
 	w := shard.BlsPublicKey{}
 	members := v.Participants()
+	currentTotalPower := numeric.ZeroDec()
 
 	for i := range members {
-		w.FromLibBLSPublicKey(members[i])
-		// isHMY := v.validatorStakes[w].isHarmonyNode
-		if v.ReadSignature(p, members[i]) == nil {
-			// TODO TODO finish this logic
+		if v.ReadSignature(p, members[i]) != nil {
+			w.FromLibBLSPublicKey(members[i])
+			currentTotalPower = currentTotalPower.Add(
+				*v.validatorStakes[w].effective,
+			)
 		}
 	}
 
-	return true
+	return currentTotalPower
```

Comment:
We should make this a persisted variable, otherwise, we have to compute it everytime a new messge is received.

Reply:
good point, the math is slow for cpu